src/bun.js/api/Timer.zig

```diff
@@ -129,6 +129,14 @@ pub const All = struct {
         return VirtualMachine.get().timer.last_id;
     }
 
+    pub export fn Bun__updateTimeoutAfterEINTR(this: *VirtualMachine, timeout: *timespec) i32 {
```

Comment:
Since you use this function thrice in the same repeating pattern I suggest wrapping that repeating logic inside e.g.

```
pub export fn Bun__updateTimeoutAfterEINTR(this: *VirtualMachine, timeout: **timespec) bool {
    if (timeout_update_ctx && timeout && timeout.* && !this.timer.getTimeout(timeout.*)) {
        timeout.* = NULL;
    }
    return !(timeout_update_ctx && timeout && timeout.* && timeout.*->tv_sec == 0 && timeout.*->tv_nsec == 0);
}
```

This would simplify callsites e.g. 
```
        if (UNLIKELY(rc == -1 && errno == EINTR)) {
            if (Bun__updateTimeoutAfterEINTR(timeout_update_ctx, timeout)) {
                continue;
            }
            loop->num_ready_polls = 0;
            break;
        }
```